Support for vnd.dovecot.pgp-encrypt

kaivol opened the pull request:
Added (rudimentary) support for the vnd.dovecot.pgp-encrypt sieve filter.

thsmi reviewed (changes requested):
Thank you very much for contributing. 

Did a rough check, everything looks good but have not tested it yet. I'll try to do that in the next days. So that this can land soon on the master. I'll also try to do a quick release afterwards.

While reviewing I stumbled upon the following "non technical" glitches:

The original author in the license header is pointing to me. But you did the work so you should add you as the initial author to all the files you created newly. If the file exists then you need to add yourself as an additional author. 

Translations are auto generated as soon as it lands in the master.  All you need to do is to add the new strings only to the en-us json file. Otherwise crowding gets confused and runs into conflicts which need to be manually resolved. I'll add a readme note to the folder containing the translations.

kaivol commented:
Thank you for your response!

Please have a look at the last changes as I am not really familiar with this project and its best practices. 
Unfortunately this also applies to the vnd.dovecot.pgp-encrypt sieve extension, and I could not find much info about it. Primarily I made sure that my implementation is compatible with the with the way mailbox.org uses the extension. 

I also updated the author in the license headers and removed the i18n strings in other languages than en-us. 
Originally, I added those because the application would complain about missing translations after adding the English translations only. Did I use the build system wrong or if not, what is the recommended workflow when introducing new translations?

yoshimo commented:
Where would i find the dovecot side of this PR and its documentation?

kaivol commented:
Unfortunately I could not find documentation or source code for the extension. 
I asked my provider ([mailbox.org](https://mailbox.org/en/)) about it, but have not yet received an answer.

kaivol commented:
According to mailbox.org, no public documentation for this plugin is available. 
It was conceived by them and developed by Dovecot (initially exclusively for mailbox.org).

thsmi commented:
Sorry completely forgot about this pull request.
